dask_pytorch/dispatch.py

```diff
@@ -0,0 +1,72 @@
+"""
+This module contains the user-facing API to submit PyTorch jobs to a Dask cluster
+"""
+
+import os
+from typing import List, Callable, Tuple
+from dask.distributed import Client
+import torch.distributed as dist
+
+
+def _get_worker_info(client: Client) -> Tuple[List[str], str]:
+    """
+    returns a list of workers (sorted), and the DNS name for the master host
+    The master is the 0th worker's host
+    """
+    workers = client.scheduler_info()["workers"]
+    worker_keys = sorted(workers.keys())
+    host = workers[worker_keys[0]]["host"]
+    return worker_keys, host
+
+
+def run(client: Client, pytorch_function: Callable, *args, **kwargs):
+    """
+    Dispatch a pytorch function over a dask cluster, and returns a list of futures
+    for the resulting tasks
+    """
+    worker_keys, host = _get_worker_info(client)
+    world_size = len(worker_keys)
+    port = 23456  # pick a free port?
+
+    futures = [
+        client.submit(
+            dispatch_with_ddp,
+            pytorch_function,
+            host,
+            port,
+            idx,
+            world_size,
+            *args,
+            workers=[w],
+            **kwargs
+        )
+        for idx, w in enumerate(worker_keys)
+    ]
+    
+    return futures
+
+
+def dispatch_with_ddp(
+    pytorch_function, master_addr, master_port, rank, world_size, backend = "nccl", *args, **kwargs
+):
```

Comment:
```suggestion
def dispatch_with_ddp(
    pytorch_function: Callable, master_addr: str, master_port: int, rank: int, world_size: int, backend: str = "nccl", *args, **kwargs
) -> Any:
```

Reply:
@jameslamb I like this idea of explicitness but I noticed when testing on my local that a lot of these typing elements need to be imported. I fixed this on the data.py one, in the future can you add those imports when making these additions in comments? (So I don't miss them :).

Reply:
oh sure, sorry about that. I guess they should have just been regular comments and not suggestions, since other lines had to be touched to make them work.